src/Compilers/CSharp/Portable/Syntax/SyntaxFactory.cs

```diff
@@ -1545,12 +1545,12 @@ public static IdentifierNameSyntax IdentifierName(string name)
         /// </summary>
         public static SyntaxTree SyntaxTree(SyntaxNode root, ParseOptions? options = null, string path = "", Encoding? encoding = null)
```

Comment:
Feels odd there isn't an overload here that allows you to specify `SourceHashAlgorithm`

Reply:
I can add it.

Reply:
Added

Reply:
The overload is no longer needed. We decided to derive custom syntax tree subclass instead.